forge-gui/src/main/java/forge/gui/card/CardDetailUtil.java

```diff
@@ -626,6 +617,12 @@ public static String composeCardText(final CardStateView state, final GameView g
                 area.append("\n");
             }
             area.append("Exiled until this leaves the battlefield: ").append(card.getUntilLeavesBattlefield());
+        } else if (card.getExiledCards() != null) {
```

Comment:
that might be two different lists, the Card may have more than one way to exile cards,
while some of them might use UntilLeavesBattlefield, while some does not

> Angel of Condemnation {2}{W}{W}
> {2}{W}, {T}: Exile another target creature. Return that card to the battlefield under its owner’s control at the beginning of the next end step.

This is in ExiledCards:

> {2}{W}, {T}, Exert Angel of Condemnation: Exile another target creature until Angel of Condemnation leaves the battlefield. (An exerted creature won’t untap during your next untap step.)

This one would show the Until Leaves Battlefield

Reply:
Hmn good point (though it also uses Remembered)

I didn't like the doubled text for cards like _Banishing Light_ but maybe it's too tricky to determine exactly

Will revert that part I guess

Reply:
Bishop of Binding for example

The ETB Trigger uses UntilLeavesBattlefield, while the Attack Trigger uses ExiledCards info